pkg/utils/utils.go

```diff
@@ -431,3 +438,51 @@ func getLinkType(ifaceStatus sriovnetworkv1.InterfaceExt) string {
 
 	return ""
 }
+
+func setVfsGuid(iface *sriovnetworkv1.Interface) error {
+	glog.Infof("setVfsGuid(): device %s", iface.PciAddress)
+	pfLink, err := netlink.LinkByName(iface.Name)
+	if err != nil {
+		glog.Errorf("setVfsGuid(): unable to get PF link for device %+v %q", iface, err)
+		return err
+	}
+	vfs, err := dputils.GetVFList(iface.PciAddress)
+	if err != nil {
+		return err
+	}
+	for _, addr := range vfs {
+		vfID, err := dputils.GetVFID(addr)
+		if err != nil {
+			glog.Errorf("setVfsGuid(): unable to get VF id %+v %q", iface.PciAddress, err)
+			return err
+		}
+		guid := generateRandomGuid()
+		if err := netlink.LinkSetVfNodeGUID(pfLink, vfID, guid); err != nil {
+			return err
+		}
+		if err := netlink.LinkSetVfPortGUID(pfLink, vfID, guid); err != nil {
+			return err
+		}
+		if err = Unbind(addr); err != nil {
+			return err
+		}
+		if err = BindDefaultDriver(addr); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func generateRandomGuid() net.HardwareAddr {
```

Comment:
This generates random guid for VFs everytime when node gets rebooted or another IB/Ether device gets configured(which result in node rebooting). Will this cause any issue in the infiniband network?

Reply:
I am not sure about, @moshe010 what do you think?

Reply:
I don't expect to be issues with that.
the PF reports its VGUIDs the SM should handle it (update/remove entries etc...).